MobiusCore/Test/EffectHandlers/EffectRouterDSLTests.swift

```diff
@@ -36,29 +36,92 @@ private enum Event: Equatable {
 class EffectRouterDSLTests: QuickSpec {
     // swiftlint:disable function_body_length
     override func spec() {
+        context("An EffectHandler which always ends as soon as it is called") {
+            var wasDisposed: Bool!
+            var connection: Connection<Int>!
+
+            beforeEach {
+                wasDisposed = false
+                connection = EffectRouter<Int, ()>()
+                    .routeEffects(equalTo: 1).to { _, response in
+                        response.end()
+                        return AnonymousDisposable {
+                            wasDisposed = true
+                        }
+                    }
+                    .asConnectable
+                    .connect { _ in }
+            }
+
+            it("should not be disposed if `end()` was called") {
+                connection.accept(1)
+                connection.dispose()
+
+                expect(wasDisposed).to(beFalse())
+            }
+
+            it("should be disposed if the connection is disposed before `end()` is called") {
```

Comment:
The description of this test case doesn't match what's being tested. I think the description is right, not sure why `wasDisposed` is false?